go-controller/pkg/util/net.go

```diff
@@ -46,21 +46,30 @@ func GetPortAddresses(portName string) (net.HardwareAddr, net.IP, error) {
 	}
 
 	// dynamic addresses have format "0a:00:00:00:00:01 192.168.1.3"
+	// dynamic addresses dual stack  "0a:00:00:00:00:01 192.168.1.3 ae70::4"
 	// static addresses have format ["0a:00:00:00:00:01 192.168.1.3"]
 	outStr := strings.Trim(out, `"[]`)
 	addresses = strings.Split(outStr, " ")
-	if len(addresses) != 2 {
+	if len(addresses) < 2 {
 		return nil, nil, fmt.Errorf("Error while obtaining addresses for %s", portName)
 	}
-	ip := net.ParseIP(addresses[1])
-	if ip == nil {
-		return nil, nil, fmt.Errorf("failed to parse logical switch port %q IP %q", portName, addresses[1])
-	}
-	mac, err := net.ParseMAC(addresses[0])
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse logical switch port %q MAC %q: %v", portName, addresses[0], err)
+	var ips []net.IP
+	var mac net.HardwareAddr
+	for i, addr := range addresses {
+		if i == 0 {
```

Comment:
this is kind of weird... I think it would be better to keep the `addresess[0]` MAC case as it is, and then iterate over `range addresses[1:]` for the IPs

Reply:
Done.